klusterletconfig/v1alpha1/types.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=klusterletconfigs
+
+// KlusterletConfig contains the configuration of a klusterlet including the upgrade strategy, config overrides, proxy configurations etc.
+type KlusterletConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of KlusterletConfig
+	// +optional
+	Spec KlusterletConfigSpec `json:"spec,omitempty"`
+
+	// Status defines the observed state of KlusterletConfig
+	// +optional
+	Status KlusterletConfigStatus `json:"status,omitempty"`
+}
+
+// KlusterletConfigSpec defines the desired state of KlusterletConfig, usually provided by the user.
+type KlusterletConfigSpec struct {
+	// Registries includes the mirror and source registries. The source registry will be replaced by the Mirror.
+	// +optional
+	Registries []Registries `json:"registries,omitempty"`
+
+	// PullSecret is the name of image pull secret.
+	// +optional
+	PullSecret corev1.ObjectReference `json:"pullSecret,omitempty"`
+
+	// NodePlacement enables explicit control over the scheduling of the agent components.
+	// If the placement is nil, the placement is not specified, it will be omitted.
+	// If the placement is an empty object, the placement will match all nodes and tolerate nothing.
+	// +optional
+	NodePlacement *NodePlacement `json:"nodePlacement,omitempty"`
```

Comment:
I think we already have nodePlacement defined in ocm api. Do you want to reuse it?

Reply:
Agree, fix with reused NodePlacement.